File: 2454-largest-local-values-in-a-matrix/2454-largest-local-values-in-a-matrix.java
class Solution {
    public int[][] largestLocal(int[][] grid) {
        int n = grid.length;
        int[][] ansMat = new int[n - 2][n - 2];
        for(int i = 0; i < n - 2; i++){
            for(int j = 0; j < n - 2; j++){
                int[] arr = {grid[i][j], grid[i][j + 1], grid[i][j + 2], grid[i + 1][j], grid[i + 1][j + 1], grid[i + 1][j + 2], grid[i + 2][j], grid[i + 2][j + 1], grid[i + 2][j + 2]};
                int max = arr[0];
                for(int k = 1; k < arr.length; k++){
                    if(arr[k] > max){
                        max = arr[k];
                    }
                }
                ansMat[i][j] = max;
            }
        }
        return ansMat;
    }
}

File: 2454-largest-local-values-in-a-matrix/2454-largest-local-values-in-a-matrix.py
class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        n = len(grid)
        ans_mat = []
        for i in range(n - 2):
            ans_mat.append([0] * (n - 2))
        for i in range(n - 2):
            for j in range(n - 2):
                s = max([grid[i][j], grid[i][j + 1], grid[i][j + 2], grid[i + 1][j], grid[i + 1][j + 1], grid[i + 1][j + 2], grid[i + 2][j], grid[i + 2][j + 1], grid[i + 2][j + 2]])
                ans_mat[i][j] = s
        return ans_mat